lib/open3.rb

```diff
@@ -482,44 +482,95 @@ class << self
   #
-  #   stdout_str, stderr_str, status = Open3.capture3([env,] cmd... [, opts])
+  # Basically a wrapper for Open3.popen3 that:
   #
-  # The arguments env, cmd and opts are passed to Open3.popen3 except
-  # <code>opts[:stdin_data]</code> and <code>opts[:binmode]</code>.  See Process.spawn.
+  # - Creates a child process, by calling Open3.popen3 with the given arguments
+  #   (except for certain entries in hash +options+; see below).
+  # - Returns as strings +stdout+ and +stderr+ the standard output
+  #   and standard error of the child process.
+  # - Returns as +status+ a <tt>Process::Status</tt> object
+  #   that represents the exit status of the child process.
   #
-  # If <code>opts[:stdin_data]</code> is specified, it is sent to the command's standard input.
+  # With no block given, returns the array <tt>[stdin_s, stdout_a, status]</tt>:
   #
-  # If <code>opts[:binmode]</code> is true, internal pipes are set to binary mode.
+  #   stdout_s, stderr_s, status = Open3.capture3('echo "Foo"')
+  #   # => ["Foo\n", "", #<Process::Status: pid 2281954 exit 0>]
+  #
+  # Like Process.spawn, this method has potential security vulnerabilities
+  # if called with untrusted input;
+  # see {Command Injection}[rdoc-ref:command_injection.rdoc].
   #
-  # Examples:
+  # Unlike Process.spawn, this method waits for the child process to exit
+  # before returning, so the caller need not do so.
   #
-  #   # dot is a command of graphviz.
-  #   graph = <<'End'
-  #     digraph g {
-  #       a -> b
-  #     }
-  #   End
-  #   drawn_graph, dot_log = Open3.capture3("dot -v", :stdin_data=>graph)
-  #
-  #   o, e, s = Open3.capture3("echo abc; sort >&2", :stdin_data=>"foo\nbar\nbaz\n")
-  #   p o #=> "abc\n"
-  #   p e #=> "bar\nbaz\nfoo\n"
-  #   p s #=> #<Process::Status: pid 32682 exit 0>
-  #
-  #   # generate a thumbnail image using the convert command of ImageMagick.
-  #   # However, if the image is really stored in a file,
-  #   # system("convert", "-thumbnail", "80", "png:#{filename}", "png:-") is better
-  #   # because of reduced memory consumption.
-  #   # But if the image is stored in a DB or generated by the gnuplot Open3.capture2 example,
-  #   # Open3.capture3 should be considered.
-  #   #
-  #   image = File.read("/usr/share/openclipart/png/animals/mammals/sheep-md-v0.1.png", :binmode=>true)
-  #   thumbnail, err, s = Open3.capture3("convert -thumbnail 80 png:- png:-", :stdin_data=>image, :binmode=>true)
-  #   if s.success?
-  #     STDOUT.binmode; print thumbnail
-  #   end
+  # Argument +options+ is a hash of options for the new process;
+  # see {Execution Options}[rdoc-ref:Process@Execution+Options].
+  #
+  # The hash +options+ is passed to method Open3.popen3;
+  # two options have local effect in method Open3.capture3:
+  #
+  # - If entry <tt>options[:stdin_data]</tt> exists,
+  #   and its string value is sent to the command's standard input:
+  #
+  #     Open3.capture3('tee', {:stdin_data => 'Foo'})
```

Comment:
Nit: a hash is not needed for keyword arguments.

```suggestion
  #     Open3.capture3('tee', stdin_data: 'Foo')
```

Reply:
Fixed.